Destroyed pods now cause a random amount of damage

OuterMostMonk opened the pull request:
If a pods integrity isn't above zero it will damage and eject all occupants upon landing

:cl: Outermonk
tweak: Makes using destroyed pods more dangerous
/:cl:

OuterMostMonk committed:
Prevents pod launch if destroyed

If a pods integrity isn't above zero it will be unable to launch until repaired

X0-11 commented:
I was considering changing this after seeing it mentioned. I left this in on purpose because, frankly, destroying the pods is too easy for such an impact. I would rather something that allows launching but makes it dangerous. Burn damage to occupants, maybe?

OuterMostMonk commented:
I do think it is a bit too easy to destroy pods. But I also think being in a ship under heavy fire should be a deadly situation. I have noticed UNSC often spawn MAC ships to harass the Covenant cruiser and simply escape through the pods the moment superstructure fails. Umbilical's are still a viable escape option (if in range) and so is repairing a pod (if not under fire).

Not too well versed in the code, but I can take a look into applying an amount of damage instead if you want.

X0-11 commented:
Simulating an incredibly rocky escape via damage application to occupants would probably be best, yeah. You'll have to fiddle with the amounts to find some that don't instantly gib a limb though

OuterMostMonk committed:
Reenables use of destroyed pods but with health risks

Destroyed pods can be deployed but the occupants will be rolling the dice as to how much damage they will take.

X0-11 commented on code/modules/halo/vehicles/types/drop_pod.dm:
```diff
@@ -139,6 +135,17 @@
+			for(var/mob/living/l in occupants) 
+				var/dam_max = BASE_VEHICLE_DEATH_EXPLODE_DAMAGE * ((bound_height / 32) + (bound_width / 32))/2 
+				l.adjustBruteLoss(dam_max/2)
+				dam_max /= 2
```
You're /2 like three times here, is that fully needed?

OuterMostMonk replied:
This code is lifted from vehiclebase.dm that it procs on_death(), same for the issue below it is the same amount of random damage that occurs to people in a vehicle when it is destroyed right? If so maybe that needs tweaking as well?

I could follow the example in vehiclebase_air.dm instead for damaging occupants of a pilotless vehicle as it is more simple.

X0-11 replied:
Ah, if it's being taken from elsewhere you can probably proc-ify the damage-dealing parts and re-use the proc here instead of copying the code verbatim.

X0-11 commented on code/modules/halo/vehicles/types/drop_pod.dm:
```diff
@@ -135,6 +135,17 @@
+				while(dam_max > 0)
+					var/dam_deal = rand(dam_max/3,dam_max)
+					dam_max -= dam_deal
+					l.adjustBruteLoss(dam_deal)
```
Subsection here's adding even more damage on? Isn't there a mob-proc for apply-damage?

OuterMostMonk committed:
Simplifies code

Instead of reusing vehicle destruction damage code, now exactly 2 rolls will be made for an amount of brute damage and burn damage each. The result should be less overall damage but with some burn instead. Also uses apply_damage as recommended

OuterMostMonk commented:
It no longer is reusing the vehicle destruction damage code, which had all sorts of things going on and could also do incredibly large amounts of damage sometimes. A benefit of using apply_damage is I have made it do a little bit of burn damage as well as brute, due to failing protection in atmospheric entry and landing badly. There is still a chance you can die or lose a limb but it is slightly less likely.

If the copied code would be better though I can go back and turn the damage element into a proc as recommended.

X0-11 commented:
It's fine as-is, if you've modified it.
